GitVersionTask/GetVersion.cs

```diff
@@ -98,36 +90,16 @@ public override bool Execute()
         {
             try
             {
-                CachedVersion versionAndBranch;
+                Tuple<CachedVersion, GitVersionContext> versionAndBranch;
```

Comment:
Not a massive fan of this, but it is a temporary hack to allow me to keep the MSBuild tasks in a working and compatible state.

With the Context being responsible for calculating effective configuration it needs to be exposed with the version for use in the variable provider.

As we get further down this refactoring hopefully more of this can all move into Core